bitswap/client/internal/session/sessionwantsender.go

```diff
@@ -209,28 +212,17 @@ func (sws *sessionWantSender) Shutdown() {
 // addChange adds a new change to the queue
 func (sws *sessionWantSender) addChange(c change) {
 	select {
-	case sws.changes <- c:
+	case sws.changes.In() <- c:
 	case <-sws.ctx.Done():
 	}
 }
 
-// addChangeNonBlocking adds a new change to the queue, using a go-routine
-// if the change blocks, so as to avoid potential deadlocks
-func (sws *sessionWantSender) addChangeNonBlocking(c change) {
-	select {
-	case sws.changes <- c:
-	default:
-		// changes channel is full, so add change in a go routine instead
-		go sws.addChange(c)
-	}
-}
-
 // collectChanges collects all the changes that have occurred since the last
 // invocation of onChange
 func (sws *sessionWantSender) collectChanges(changes []change) []change {
-	for len(changes) < changesBufferSize {
+	for range sws.changes.Len() {
 		select {
-		case next := <-sws.changes:
+		case next := <-sws.changes.Out():
```

Comment:
@gammazero you wrote the library so you probably know better.

Are there any benefits of doing like above?
```go
changes := sws.changes.Out()
for range sws.changes.Len() {
	select {
	case next := <-changes:
...
```